lib/collection/src/optimizers_builder.rs

```diff
@@ -71,8 +70,12 @@ pub struct OptimizersConfig {
     pub indexing_threshold: Option<usize>,
     /// Minimum interval between forced flushes.
     pub flush_interval_sec: u64,
-    /// Maximum available threads for optimization workers
-    pub max_optimization_threads: usize,
+    /// Max number of threads (jobs) for running optimizations per shard.
+    /// Note: each optimization job will also use `max_indexing_threads` threads by itself for index building.
+    /// If null - have no limit and choose dynamically to saturate CPU.
+    /// If 0 - no optimization threads, optimizations will be disabled.
```

Comment:
Do you have a different suggestion? We need to support both options.

I'd argue that using `0` for `auto` in other places is wrong, and that we should (have) use(d) `null` there instead.

Reply:
One solution can be an enum so that users can set number or "auto" string. I am mostly worried about the case of deserialization where it's hard to capture the semantic of `{ "field": null }` vs `{ }`, which has been the problem in other places